TechnicalDocuments/Proposals/BuildersToFunctions.md

```diff
@@ -0,0 +1,158 @@
+    }
+}
+```
+
+Which would allow the consumer to set a default parameter value in the `FlowController` initializer. Instead of:
+
+```swift
+struct ClinicalRecordsFlowController {
+    private let navigation: Flow
+    private let modal: Flow
+    private let presenting: Flow
+    private let builders: ClinicalRecordsChildBuilders
+
+    public init(navigation: Flow,
+                modal: Flow,
+                presenting: Flow,
+                builders: ClinicalRecordsChildBuilders) {
+        self.navigation = navigation
+        self.modal = modal
+        self.presenting = presenting
+        self.builders = builders
+    }
+```
+
+One would:
+
+```swift
+struct ClinicalRecordsFlowController {
+    private let navigation: Flow
+    private let modal: Flow
+    private let presenting: Flow
+
+    public init(navigation: Flow,
+                modal: Flow,
+                presenting: Flow) {
+        self.navigation = navigation
+        self.modal = modal
+        self.presenting = presenting
+    }
+```
+
+Particular medical histories (e.g. for minors) can also be created, but in this case, the minor session would be passed instead of relying on `Current`. This situation (with child builders) can be observed at `ClinicalRecordsChildBuilders.swift`:
+
+```swift
+extension Builder {
+    struct ClinicalRecords {
+        static func makeMedicalHistory(with session: Session, flow: Flow) -> UIViewController 
+        static func makeMedicalHistory(flow: Flow) -> UIViewController
+    }
+}
+```
+
+This would remove the intricate structure we have with builders and their children and protocols.
+
+## Impact on an existing codebas
+
+There is the question about moving `UserSession`, or something in that shape, to `Current`. Based on an initial conversation with Anders, and quoting him, it would be a "fairly low" effort. This is for me a point that should be thought about carefully and could be a reason to reject this proposal. 
+
+As a proposal on its own, it wouldn't have any immediate impact. For new features, and assuming `UserSession` in `Current` is in place, those could start using this approach. **Builders that don't rely on `UserSession` could be migrated immediately**. 
+
+## Alternatives considered
+
+Using Ilya's suggestion about structs with functions as properties, versus static free functions. The overall goals of the proposal are preserved and it provides flexibility for testing. 
+
+---
+
+Decision made during the call:
+
+1. Popular Vote: Use/Move UserSession (not the entity, but what it represents), to `Current`. **Majority has approved.**
+2. Popular Vote: In favour of the original proposal to use static builder methods. **Majority has approved.**
+   1. Amendment to the original proposal: **by default** `FlowControllers` will only be injected with `Flows`, but if testing is deemed, then further dependencies can be passed.
+     
```

Comment:
I see `analytics` being often triggered in flowControllers. 
And along with this, (if it's still an assumption) just `Builder`s can access `Current`, so then analytics can be injected to a flowController.

Reply:
analytics should be in `ViewModels` not FlowControllers.